### DISCUSSION TITLE
RequestValidationError is **not** a sub-class of Pydantic's ValidationError

### DISCUSSION THREAD

**Discussion by valentinoli** [fastapi/fastapi#10421] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.exception_handler(ValidationError)
async def validation_exception_handler(request: Request, exc: ValidationError):
    return PlainTextResponse(
        str(exc), status_code=status.HTTP_500_INTERNAL_SERVER_ERROR
    )
```

### Description

I was trying to handle both `RequestValidationError` and `ValidationError` by writing a custom handler for `ValidationError` but that doesn't work.

[The docs](https://fastapi.tiangolo.com/tutorial/handling-errors/#override-request-validation-exceptions) state that 

> RequestValidationError is a sub-class of Pydantic's [ValidationError](https://pydantic-docs.helpmanual.io/usage/models/#error-handling)

However, looking at [FastAPI source code](https://github.com/tiangolo/fastapi/blob/c1adce4fe93a0035e69988f7e051cfab97d8acef/fastapi/exceptions.py#L36) reveals that this is not the case.

### Operating System

Linux

### FastAPI Version

0.101.0

### Pydantic Version

2.1.1

### Python Version

3.11.3

**Comment by Kludex** [comment, marked as answer]
- I've created https://github.com/tiangolo/fastapi/issues/10786 from this discussion. Thanks.
